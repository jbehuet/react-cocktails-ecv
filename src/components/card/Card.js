import { LikeButton, RefreshButton, RemoveButton } from '../buttons';
import { Loader } from '../loader/Loader';
import './Card.css';

const STR_INGREDIENT = "strIngredient";

export function Card({ cocktail, onLike, onRetry, onRemove, isLoading }) {
    const getIngredients = () => {
        return Object.keys(cocktail)
            .filter(key => key.startsWith(STR_INGREDIENT))
            .filter(key => !!cocktail[key])
            .map((key) => {
                const idx = key.split(STR_INGREDIENT)[1];
                return {
                    ingredient: cocktail[key],
                    measure: cocktail[`strMeasure${idx}`] ? cocktail[`strMeasure${idx}`] : ''
                }
            })
    }

    return (
        <div className="card">
            <div className="card-header">
                {!isLoading && <img src={cocktail.strDrinkThumb} alt={cocktail.strDrink} />}
            </div>

            <div className="card-body">
                <div className="text">
                    {isLoading ?
                        <Loader />
                        :
                        <>
                            <h1>{cocktail.strDrink}</h1>
                            <span className="badge">{cocktail.strCategory}</span>
                            <div className="soft">{cocktail.strAlcoholic !== 'Alcoholic' && "SOFT"}</div>
                            <ul className="ingredients-list">
                                {getIngredients().map(i =>
                                    <li key={i.ingredient}>{`${i.measure} ${i.ingredient}`}</li>
                                )}
                            </ul>
                        </>
                    }
                </div>
            </div>
            {onLike && <LikeButton onClick={() => onLike(cocktail)} />}
            {onRetry && <RefreshButton onClick={onRetry} />}
            {onRemove && <RemoveButton onClick={() => onRemove(cocktail)} />}
        </div>
    )

}