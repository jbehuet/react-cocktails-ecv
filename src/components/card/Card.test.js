import React from "react";
import { create } from "react-test-renderer";

import { Card } from "./Card";

const fakeCocktail = {
    "idDrink": "13190",
    "strDrink": "Kool-Aid Shot",
    "strCategory": "Shot",
    "strAlcoholic": "Alcoholic",
    "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/fegm621503564966.jpg",
    "strIngredient1": "Vodka",
    "strMeasure1": "1 shot ",
}

describe("Card component", () => {

    it("renders correctly when is Loading", () => {
        const component = create(<Card isLoading={true} onLike={() => { }} onRetry={() => { }} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it("renders correctly with like button and retry", () => {
        const component = create(<Card cocktail={fakeCocktail} onLike={() => { }} onRetry={() => { }} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it("renders correctly with remove button", () => {
        const component = create(<Card cocktail={fakeCocktail} onRemove={() => { }}/>).toJSON();
        expect(component).toMatchSnapshot();
    });
});
