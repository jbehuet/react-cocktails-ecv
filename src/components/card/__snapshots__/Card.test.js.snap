// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly when is Loading 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  />
  <div
    className="card-body"
  >
    <div
      className="text"
    >
      <div
        className="loader"
      >
        Loading...
      </div>
    </div>
  </div>
  <button
    className="like"
    onClick={[Function]}
  >
    <i
      className="fas fa-heart"
    />
  </button>
  <button
    className="refresh"
    onClick={[Function]}
  >
    <i
      className="fas fa-redo-alt"
    />
  </button>
</div>
`;

exports[`Card component renders correctly with like button and retry 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <img
      alt="Kool-Aid Shot"
      src="https://www.thecocktaildb.com/images/media/drink/fegm621503564966.jpg"
    />
  </div>
  <div
    className="card-body"
  >
    <div
      className="text"
    >
      <h1>
        Kool-Aid Shot
      </h1>
      <span
        className="badge"
      >
        Shot
      </span>
      <div
        className="soft"
      />
      <ul
        className="ingredients-list"
      >
        <li>
          1 shot  Vodka
        </li>
      </ul>
    </div>
  </div>
  <button
    className="like"
    onClick={[Function]}
  >
    <i
      className="fas fa-heart"
    />
  </button>
  <button
    className="refresh"
    onClick={[Function]}
  >
    <i
      className="fas fa-redo-alt"
    />
  </button>
</div>
`;

exports[`Card component renders correctly with remove button 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <img
      alt="Kool-Aid Shot"
      src="https://www.thecocktaildb.com/images/media/drink/fegm621503564966.jpg"
    />
  </div>
  <div
    className="card-body"
  >
    <div
      className="text"
    >
      <h1>
        Kool-Aid Shot
      </h1>
      <span
        className="badge"
      >
        Shot
      </span>
      <div
        className="soft"
      />
      <ul
        className="ingredients-list"
      >
        <li>
          1 shot  Vodka
        </li>
      </ul>
    </div>
  </div>
  <button
    className="remove"
    onClick={[Function]}
  >
    <i
      className="fas fa-trash"
    />
  </button>
</div>
`;
